File: base/src/main/java/com/appoets/base/base/BaseViewModel.java
package com.appoets.base.base;

import java.lang.ref.WeakReference;

import androidx.lifecycle.ViewModel;

public abstract class BaseViewModel<N> extends ViewModel {
    private WeakReference<N> mNavigator;

//    @Inject
//    BaseRespositary respositary;

    protected N getNavigator() {
        return mNavigator.get();
    }

    public void setNavigator(N navigator) {
        this.mNavigator = new WeakReference<>(navigator);
    }
}

File: base/src/main/java/com/xjek/base/base/BaseViewModel.kt
package com.xjek.base.base

import androidx.lifecycle.ViewModel
import io.reactivex.disposables.CompositeDisposable
import java.lang.ref.WeakReference

open class BaseViewModel<N> : ViewModel() {

    private var compositeDisposable = CompositeDisposable()
    private lateinit var mNavigator: WeakReference<N>

    var navigator: N
        get() = mNavigator.get()!!
        set(navigator) {
            this.mNavigator = WeakReference(navigator)
        }

    fun getCompositeDisposable() = compositeDisposable

    override fun onCleared() {
        super.onCleared()
        compositeDisposable.clear()
    }
}
